src/backend/api/reviews.js

```diff
@@ -0,0 +1,117 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+const reviewsColumns = new Set([
+  "title",
+  "description",
+  "meal_id",
+  "stars",
+  "created_date",
+]);
+
+// this function checks if input data valid and makes possible to update only neccesary data in table
+function checkTableData(table) {
+  try {
+    let validTableData = {};
+    for (const key in table) {
+      if (reviewsColumns.has(key)) {
+        validTableData[key] = table[key];
+      }
+    }
+    if ("meal_id" in validTableData) {
+      if (isNaN(Number(validTableData.meal_id))) {
+        throw new Error();
+      }
+    }
+
+    if ("stars" in validTableData) {
+      if (isNaN(Number(validTableData.stars))) {
+        throw new Error();
+      }
+    }
+    if ("created_date" in validTableData) {
+      const createdDateFormatted = new Date(validTableData.created_date);
+      if (!Date.parse(createdDateFormatted)) {
+        throw new Error();
+      }
+    }
+    return validTableData;
+  } catch (error) {
+    throw error;
```

Comment:
Nitpick(error handling): 
First of, none of the code above actually throws an error on it's own so there is no try/catch for the code itself.
So the only thing you would catch is the error you yourself throw which can be fine sometimes, but here you simply catch and re-throw. In such cases, it's better to just let the error bubble up and be caught somewhere else.

Now you might make this more valuable by also logging the error message to the console before throwing it (so you can debug easier), or perhaps throw a new error that is more suitable for the user of the API to read.